### PR TITLE
CI fix: Fix occasional port bind error happening in mock mining

### PR DESCRIPTION
**Pull request by jferrant**
Check if in feature == testing and reserve test observer port in USED_PORTS

### CODE REVIEW

File: testnet/stacks-node/src/tests/mod.rs [hunk 1]
```diff
@@ -109,11 +110,15 @@ pub fn gen_random_port() -> u16 {
...
     let range_len = (1024..u16::MAX).len();
     loop {
+        // Note it needs to be +1 because we reserve one port for the event observer
         assert!(
-            USED_PORTS.lock().unwrap().len() < range_len,
+            USED_PORTS.lock().unwrap().len() + 1 < range_len,
             "No more available ports"
         );
         let port = rng.gen_range(1024..u16::MAX); // use a non-privileged port between 1024 and 65534
+        if port == EVENT_OBSERVER_PORT {
```
Comment (obycode) [inline testnet/stacks-node/src/tests/mod.rs:119]: Should we just initialize `USED_PORTS` to include the `EVENT_OBSERVER_PORT` instead?

Comment (jferrant) [reply testnet/stacks-node/src/tests/mod.rs]: Yes. :) Done!

**Review by hstove — APPROVED** [review]
smart catch! LGTM

**Review by hstove — APPROVED** [review]

**Review by obycode — APPROVED** [review]
LGTM!

**Pull request merged**